benchmarks/conftest.py

```diff
@@ -0,0 +1,71 @@
+from attr import dataclass
+import json
+import pytest
+from typing import List
+
+from ariadne import gql, make_executable_schema, QueryType
+
+
+@dataclass
+class Group:
+    name: str
+    roles: list
+
+
+@dataclass
+class Avatar:
+    size: int
+    url: str
+
+
+@dataclass
+class User:
+    id: int
+    name: str
+    group: Group
+    avatar: List[Avatar]
+
+
+with open("benchmarks/data.json") as f:
+    data = json.load(f)
+
+
+def users_data():
+    for row in data["users"]:
+        yield User(**row)
```

Comment:
Remember that `dataclass` creation is not recursive. You need to manually hydrate Group and Avatars.

Reply:
Done, used "get" instead of "pop". Pop worked only for first test ( no matter which were situated in file as first)